restx-specs-tests/src/main/java/restx/tests/RestxSpecTestsRunner.java

```diff
@@ -48,7 +50,10 @@ public RestxSpecTestsRunner(Class<?> testClass) throws InitializationError, Ille
         super(testClass);
         FindSpecsIn findSpecsIn = getTestClass().getJavaClass().getAnnotation(FindSpecsIn.class);
         if (findSpecsIn != null) {
-            tests = new RestxSpecTests(new RestxSpecRule(findSpecsIn.withRouterPath()), RestxSpecTests.findSpecsIn(findSpecsIn.value()));
+            Factory factory = Factory.newInstance();
+            WebServerSupplier supplier = factory.queryByClass(WebServerSupplier.class).findOne().get().getComponent();
+            RestxSpecRule rule = new RestxSpecRule(findSpecsIn.withRouterPath(), supplier, factory);
+            tests = new RestxSpecTests(rule, RestxSpecTests.findSpecsIn(findSpecsIn.value()));
```

Comment:
This merge request is not for java 11. These tests are failing with java 8. I have not tested them with a jdk11 to be honest, I first wanted to make the tests pass on java 8.

I provide elements in [my second comment](https://github.com/restx/restx/issues/312#issuecomment-612996458) in #312:

>It appears that GivenJongoCollectionSpecRuleSupplier is instantiated only once when executing the Maven job, at the first execution of a test running with RestxSpecTestsRunner or MongoRestxSpecTestsRunner.
>
>In my case, the first one was ParametersResourceSpecTest, which is not using MongoRestxSpecTestsRunner. So without an embedded Mongo being launched, the value in the settings is set to it's default value: localhost:21017, and stay like this for the whole testing suite.
>
>Because embedded Mongo is started for every class, we must refresh these settings for every test class, I had to create a new Factory instance for every RestxSpecTests (see: nhumblot@922ec78)

Reply:
Thanks, I missed this comment.
So I'm ok with this change on the runner.

Reply:
Ok, thank you. I will reply to the other two comments also and try to give as many details as I can. I will come back to you when I get the time to write them. :)